pkg/client/keycloak/config.go

```diff
@@ -184,6 +187,7 @@ func (kc *KeycloakConfig) ReadFiles() error {
 		}
 	}
 
+	validateSSOProviderRealm(kc.SelectSSOProvider)
```

Comment:
I think we can have a separate `.Validate` method on the ConfigModule.
This way, it'll be validated automatically by the config module setup.

Something like this: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/8df317f8d17c5618e651300e444dc042b0e6d3c6/internal/kafka/internal/config/kafka.go#L79-L81

This way, the `.ReadFiles()` does not have two responsibilities
1. to read a file
2. perform validation